src/librustc/hir/intravisit.rs

```diff
@@ -404,7 +404,7 @@ pub fn walk_local<'v, V: Visitor<'v>>(visitor: &mut V, local: &'v Local) {
     // Intentionally visiting the expr first - the initialization expr
     // dominates the local's definition.
     walk_list!(visitor, visit_expr, &local.init);
-
+    walk_list!(visitor, visit_attribute, local.attrs.iter());
```

Comment:
I would almost recommend an audit of the entire `intravisit` module to ensure that anywhere attributes can appear they're being visited. This missing line took me three hours to debug because I assumed it was something wrong with _my_ code.

Reply:
It looks like every type in `hir` that has an `attrs` field is now being walked by `intravisit`. I think I've fixed the last of the discrepancies.

Reply:
On a different note, this field and a couple others use `ThinVec` while the `attrs` fields on all the other types use `HirVec`. I'm not sure of the difference but I think it's just an alias anyway. However, I got an error when passing `&local.attrs` here when it works fine for invocations with `HirVec`.